packages/utilities/fast-colors/src/color-converters.ts

```diff
@@ -78,24 +78,51 @@ export function contrastRatio(a: ColorRGBA64, b: ColorRGBA64): number {
  * Calculate an overlay color that uses rgba (rgb + alpha) that matches the appareance of a given solid color when placed on the same background
  * @param rgbMatch - The solid color the overlay should match in appearance when placed over the rgbBackground
  * @param rgbBackground - The background on which the overlay rests
- * @param rgbOverlay - The rgb color of the overlay. Typically this is either pure white or pure black. This color will be used in the returned output
+ * @param rgbOverlay - The rgb color of the overlay. Typically this is either pure white or pure black and when not provided will be determined automatically. This color will be used in the returned output
  * @returns The rgba (rgb + alpha) color of the overlay
  *
  * @public
  */
 export function calculateOverlayColor(
     rgbMatch: ColorRGBA64,
     rgbBackground: ColorRGBA64,
-    rgbOverlay: ColorRGBA64
+    rgbOverlay: ColorRGBA64 = null
 ): ColorRGBA64 {
-    const rChannel: number =
-        (rgbMatch.r - rgbBackground.r) / (rgbOverlay.r - rgbBackground.r);
-    const gChannel: number =
-        (rgbMatch.g - rgbBackground.g) / (rgbOverlay.g - rgbBackground.g);
-    const bChannel: number =
-        (rgbMatch.b - rgbBackground.b) / (rgbOverlay.b - rgbBackground.b);
-    const alpha: number = (rChannel + gChannel + bChannel) / 3;
-    return new ColorRGBA64(rgbOverlay.r, rgbOverlay.g, rgbOverlay.b, alpha);
+    function calcChannel(match: number, background: number, overlay: number): number {
+        if (overlay - background === 0) {
+            return 0;
+        } else {
+            return (match - background) / (overlay - background);
+        }
+    }
+
+    function calcRgb(
```

Comment:
And reason this function can't be extracted to the outer scope so it doesn't need to be re-defined every time the outer function runs?